### DISCUSSION TITLE
Amazing project

### DISCUSSION THREAD

**Discussion by alros100** [Builty/TonexOneController#34] [General]
Hi 
I have been looking for a solution for control of ToneX one. Amazing!

Questions:
I have three esp32-S3 wroom development boards hanging around. Would they work or only the waveshare S3 zero/ mini boards?

I will start with one, but I am also hoping to build two with direct midi wiring so i can run two amp/ cabs at same time from left and right send / receive loops and just change the presets .

Is the direct midi firmware up already ?

Thanks so much!
Al

**Comment by Builty** [comment]
Hi Al. You'll need to check the schematic of those boards, and see if they support USB host. Have a look at the power circuit, and see if it allows 5v to be connected somewhere other than the USB port, and if that voltage does also go to the USB port.
Most of the Waveshare boards are OK here, where some of the boards like from Seeed Studio are not OK (diode blocks power.)

If the above is OK, then the other factor is how much flash and PSRAM. There are lots of different variants.
The Zero board is 4 MB flash, quad interface, and quad mode PSRAM.
The 4.3B is 8 MB flash, octal interface, and octal interface PSRAM.

If you run a pre-built release, and any of these settings don't match yours, it will crash on boot. If you can compile the code however, then its not too difficult to change the settings to match.

Regarding wired Midi, the pre-built releases 1.0.3.2 don't support it, but I hope to have it finished in the next few days. Its already working well on the Zero and the 1.69 board. Just need to get the 4.3B finished.

**Reply by alros100** [reply]
Awesome thanks so much - ordered some zeros just in case and very much looking to test this out 

Amazing you are able to develop something like this for the community ! 

Al

**Reply by alros100** [reply]
Hi Builty

I just saw your YouTube video on the 169 board - super cool! Will it be possible to use it with direct midi as well ? 

In fact. As a follow up, do these ESP32-S3 boards allow your firmware to support connecting footswitch controller, direct midi, and the screen at the same time ?

Looking forward to testing this - have some zero and 169 boards arriving !

Thanks again for a great project

Al

**Reply by Builty** [reply]
Hi Al,
Thanks. Yes, the 1.69" board supports direct Midi. And also yes, you can use all of the options at the same time if you like.
The 1.0.4.1 release candidate builds in the Releases folder here support all of these options.
I just need to finish a few wiring diagrams.

**Reply by alros100** [reply]
Ok - Thanks So much

**Reply by alros100** [reply]
![image](https://github.com/user-attachments/assets/214e1d1a-fa45-4cd6-8434-58f0a10d28a4)

Looking good !

Al

**Reply by alros100** [reply]
Hi Builty

Just wanted to report that I fully set up the 169 board with external midi and the foot switch and it works perfectly ! I tested midi control with both an FM9 and GT-1000

Thanks so much! Donated for the cause.

The only thing I noticed is some noise/ chatter from the 169's onboard buzzer.

Al

**Reply by Builty** [reply]
Great, thank you.
Another user has reported some noise from the buzzer also. Does your PCB have a "V2" sticker on the USB-C port?

**Reply by alros100** [reply]
Hi

No sticker on mine. I can't see a version code elsewhere. I got it off Amazon. Here is a picture.  

Ps. I put a little precautionary heat sink on the S3 as it seemed a little hot. 
![image](https://github.com/user-attachments/assets/b2c09976-3edb-4e08-beed-ccea5fd7a0c9)

**Comment by Builty** [comment]
OK so its a V1, makes sense. I've got a tech support request in with Waveshare about options to fix this.

Could you try blocking this little hole shown below, and see how much quieter that makes it?

![image](https://github.com/user-attachments/assets/d90f679f-7e6e-42af-8c3a-996c0b37b815)

**Comment by Builty** [comment]
Refer here for more discussions on the V1 noisy buzzer issue:
https://github.com/Builty/TonexOneController/issues/37

**Reply by alros100** [reply]
Thanks so much 
Al
